src/lib/Libwin/passwd.c

```diff
@@ -4132,17 +4097,24 @@ wsystem(char *cmdline, HANDLE user_handle)
 	}
 
 	run_exit = GetLastError();
+	if (rc == 0) {
+		log_err(-1, __func__, "failed in process creation");
+	}
 
 	/* restore current working directory */
 	if (changed_dir)
 		chdir(current_dir);
 
 	if (rc) { /* CreateProcess* successful */
+		stat = WaitForSingleObject(pi.hProcess, INFINITE);
+		if (stat != WAIT_OBJECT_0) {
+			log_eventf(PBSEVENT_ERROR, 0, LOG_DEBUG, __func__, "failed in WaitForSingleObject, with errno %d", stat);
```

Comment:
should we use log_err(-1,) when stat == WAIT_FAILED. from https://docs.microsoft.com/en-us/windows/win32/api/synchapi/nf-synchapi-waitforsingleobject

> 

WAIT_FAILED(DWORD)0xFFFFFFFF | The function has failed. To get extended error information, call GetLastError.
-- | --